bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/AutoScaling.java

```diff
@@ -0,0 +1,143 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Yatta and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Yatta - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.swt;
+
+import java.util.*;
+
+import org.eclipse.swt.internal.*;
+
+/**
+ * Utility for checking and enabling monitor-specific auto scaling.
+ * @since 3.132
+ */
+public final class AutoScaling {
+	/**
+	 * System property to enable to scale the application on runtime
+	 * when a DPI change is detected.
+	 * <ul>
+	 * <li>"true": the application is scaled on DPI changes</li>
+	 * <li>"false": the application will remain in its initial scaling</li>
+	 * </ul>
+	 * <b>Important:</b> This flag is only parsed and used on Win32. Setting it to
+	 * true on GTK or cocoa will be ignored.
+	 */
+	private static final String SWT_AUTOSCALE_UPDATE_ON_RUNTIME = "swt.autoScale.updateOnRuntime";
+	private static final String SWT_AUTOSCALE = "swt.autoScale";
+	private static final String SWT_AUTOSCALE_DISABLE_COMPATIBILITY_CHECK = "swt.autoScale.force";
+	private static String autoScaleValue;
+	static {
+		autoScaleValue = System.getProperty (SWT_AUTOSCALE);
+	}
+	private static final Set<String> ALLOWED_VALUES = Set.of("false", "quarter", "exact");
```

Comment:
In my opinion, we should even remove the "false" option. And the variable should have a more specific name, as is's unclear for what these values are allowed.
```suggestion
	private static final Set<String> ALLOWED_AUTOSCALE_VALUES_FOR_UPDATE_ON_RUNTIME = Set.of("quarter", "exact");
```